tests/FSharp.Test.Utilities/Compiler.fs

```diff
@@ -1219,6 +1270,64 @@ module rec Compiler =
             withErrors [expectedError] result
 
+        module StructuredResultsAsserts = 
+            type SimpleErrorInfo =    
+                { Error:   ErrorType
+                  Range:   Range        
+                  Message: string }
+
+            let withResults (expectedResults: SimpleErrorInfo list) result : CompilationResult =
+                let mappedResults = expectedResults |> List.map (fun s -> { Error = s.Error;Range = s.Range;  Message = s.Message; NativeRange = Unchecked.defaultof<_>; SubCategory = ""})
+                Compiler.Assertions.withResultsIgnoreNativeRange mappedResults result
+
+            let withResult (expectedResult: SimpleErrorInfo ) (result: CompilationResult) : CompilationResult =
+                withResults [expectedResult] result
+
+
+
+        module TextBasedDiagnosticAsserts = 
+            open FSharp.Compiler.Text.Range
+
+            let private messageAndNumber erroryType = 
+                match erroryType with
+                | ErrorType.Error n -> "error",n
+                | ErrorType.Warning n-> "warning",n
+                | ErrorType.Hidden n
+                | ErrorType.Information n-> "info",n
+
+            let normalizeNewLines (s:string) = s.Replace("\r\n","\n").Replace("\n",Environment.NewLine) 
+
+            let private renderToString (cr:CompilationResult) = 
+                [ for (file,err) in cr.Output.PerFileErrors do
+                    let m = err.NativeRange
+                    let file = file.Replace("/", "\\")
+                    let severity,no = messageAndNumber err.Error
+                    let adjustedMessage = err.Message |> normalizeNewLines
+                    let location = 
+                        if  (equals m range0) || (equals m rangeStartup) || (equals m rangeCmdArgs) then
+                            ""
+                        else 
+                            // The baseline .bsl files use 1-based notation for columns, hence the +1's
+                            sprintf "%s(%d,%d,%d,%d):" file m.StartLine (m.StartColumn+1) m.EndLine (m.EndColumn+1)
+                    Environment.NewLine + $"{location} {err.SubCategory} {severity} FS%04d{no}: {adjustedMessage}" + Environment.NewLine
+                ]
+                |> String.Concat
+
+            let withResultsMatchingFile (path:string) (result:CompilationResult) = 
+                let expectedContent = File.ReadAllText(path) |> normalizeNewLines
+                let actualErrors = renderToString result                
+
+                match Environment.GetEnvironmentVariable("TEST_UPDATE_BSL") with
+                | null -> ()
+                | _ when expectedContent = actualErrors -> ()
+                | _ -> File.WriteAllText(path, actualErrors)
+
+                match Assert.shouldBeSameMultilineStringSets expectedContent actualErrors with
+                | None -> () // Assert.That(actualErrors, Is.EqualTo(expectedContent).NoClip,path)
```

Comment:
The comment is confusing here, I'd either explain things further or just remove the comment.